ee/clickhouse/views/test/test_clickhouse_experiments.py

```diff
@@ -309,6 +309,256 @@ def test_transferring_holdout_to_another_group(self):
             [{"properties": [], "rollout_percentage": 5, "variant": f"holdout-{holdout_2_id}"}],
         )
 
+    def test_saved_metrics(self):
+        response = self.client.post(
+            f"/api/projects/{self.team.id}/experiment_saved_metrics/",
+            {
+                "name": "Test Experiment saved metric",
+                "description": "Test description",
+                "query": {"events": [{"id": "$pageview"}]},
```

Comment:
This should only accept `ExperimentTrendsQuery | ExperimentFunnelsQuery`

(fine to do the validation in a follow up, but maybe better to pass the correct query now to avoid having to update this later)

Reply:
oh I see! How do you generally validate this? I could just check the `kind` in the json, anything else I can do with hogql?

Reply:
(not important in this test will do it in a follow up 👍 )

Reply:
would something like this work?

```
if value["kind"] === "ExperimentTrendsQuery":
	ExperimentTrendsQuery(**value)
elif value["kind"] === "ExperimentFunnelsQuery":
	ExperimentFunnelsQuery(**value)
else:
	raise
```

Reply:
ah I see we have pydantic generated schema, should be able to validate that, will do that 👌

Reply:
`metrics` need similar validation right? Not sure where this is happening right now, don't see it in the viewset / serializer

Reply:
yep, that would be great!

actually, do we validate `filters` at all? I don't see it in the serializer

Reply:
yeah a little bit, shouldn't have `properties` and should exist is the only validation.